sa/saro.go

```diff
@@ -1289,62 +1289,120 @@ func (ssa *SQLStorageAuthority) SerialsForIncident(req *sapb.SerialsForIncidentR
 func (ssa *SQLStorageAuthorityRO) GetRevokedCerts(req *sapb.GetRevokedCertsRequest, stream sapb.StorageAuthorityReadOnly_GetRevokedCertsServer) error {
 	atTime := time.Unix(0, req.RevokedBeforeNS)
 
-	clauses := `
-		WHERE notAfter >= ?
-		AND notAfter < ?
-		AND issuerID = ?
-		AND status = ?`
-	params := []interface{}{
-		time.Unix(0, req.ExpiresAfterNS),
-		time.Unix(0, req.ExpiresBeforeNS),
-		req.IssuerNameID,
-		core.OCSPStatusRevoked,
-	}
-
-	selector, err := db.NewMappedSelector[crlEntryModel](ssa.dbReadOnlyMap)
-	if err != nil {
-		return fmt.Errorf("initializing db map: %w", err)
-	}
+	if req.ShardIdx != 0 {
```

Comment:
This deserves to be split out into separate named functions: `getRevokedCertsFromCertificateStatus` / `getRevokedCertsFromRevokedCertificatesTable`